std/typecons.d

```diff
@@ -6896,3 +6896,418 @@ unittest
         ubyte, ubyte, T3, T3,
     );
 }
+
+private enum isNullable(T) = is(typeof(T.init is null));
```

Comment:
Could these be moved to std.traits? If they are helpful here then the chances of them being helpful to someone else is high.